### PR TITLE
feat(route): add route of Yamap

### PR DESCRIPTION
**Pull request by valuex**
## Involved Issue / 该 PR 相关 Issue

Close #

## Example for the Proposed Route(s) / 路由地址示例

```routes
/yamap
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [ X] New Route / 新的路由
  - [ ] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Anti-bot or rate limit / 反爬/频率限制
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [ ] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [ ] Parsed / 可以解析
  - [ ] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

### CODE REVIEW

**Comment by valuex** [comment]
Thanks for your kind code review with patience.  Most has been fixed except for one:  
 (https://github.com/DIYgod/RSSHub/pull/16902#discussion_r1773641585).    
It would be highly appreciated if any further explanation could be given.  
Thanks.

File: lib/routes/yamap/articles.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+            catch { mountainName = 'Japan'; }
+            return mountainName;
+        },
+    }));
+
+
+    const items = await Promise.all(
+        lists.map((item) =>
+            cache.tryGet(item.link, async () => {
+                const response = await got.get(item.link);
```
Comment (TonyRL) [inline lib/routes/yamap/articles.ts:47]: https://github.com/DIYgod/RSSHub/pull/16902#discussion_r1773633082

File: lib/routes/szftedu/dongtai.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
+import { Route } from '@/types';
```
Comment (TonyRL) [inline lib/routes/szftedu/dongtai.ts:1]: You're submitting PR for namespace `yamap`. Please remove files related to other namespace, i.e., `szftedu` from the PR.

File: lib/routes/yamap/articles.ts (hunk 1 above)
Comment (valuex) [reply lib/routes/yamap/articles.ts]: Revised. Thanks.

**Pull request merged**